torch/fx/experimental/Partitioner.py

```diff
@@ -393,3 +439,86 @@ def get_bfs_level_partition(self) -> None:
                 next_level = set()
                 level += 1
         return
+
+    def is_embedding_node(self, node: Node) -> bool:
+        """Check if a node is an embedding node"""
+        if node.op == 'call_module':
+            submodule_dict = dict(self.graph_module.named_modules())
+            submodule = submodule_dict[node.target]
+            if 'Embedding' in str(submodule):
+                return True
+        return False
+
+    def sparse_nn_partition(self, available_mem_bytes: Union[float, int]) -> None:
+        """This method partition a sparse nn module.
+           It first traverse all the nodes and do the partitions based on memory size.
+           If the current partition has no enough memory left for a new op node
+           (call_module, call_method, call_function), a new partition is created.
+           Different for size_based_partition, when traversing cross the boundary between
+           non-embedding nodes and embedding nodes, a new partition is created regardlessly.
+           For example, if the current node is a non-embedding node but the next node is an
+           embedding node, a new partition is created for the next node.
+           After the partition, the partitions are combined as much as possible.
+           The rule is that non-embedding partition only combines non-embedding ones.
+           So as the embedding ones.
+        """
+        # Track memory size for embedding partitions and non-embedding partitions separately
+        embedding_partition_to_used_mem_bytes: Dict[Partition, int] = {}
+        non_embedding_partition_to_used_mem_bytes: Dict[Partition, int] = {}
+        # A Flag to check the boundary
+        partition_embeddings: bool = False
+        partition = self.create_partition()
+        used_mem_bytes = 0
+        for node in self.graph_module.graph.nodes:
+            if node.op in {'call_module', 'call_method', 'call_function'}:
+                # Check if crossing the boundary between embedding nodes and non embedding nodes
+                if (self.is_embedding_node(node) and (not partition_embeddings)) \
+                        or ((not self.is_embedding_node(node)) and partition_embeddings):
+                    # Crossing the boundary
+                    # Check if the current partition is an empty partition
+                    if used_mem_bytes != 0:
+                        # The current partition isn't an empty partition. Create a new one.
+                        if partition_embeddings:
+                            embedding_partition_to_used_mem_bytes[partition] = used_mem_bytes
+                        else:
+                            non_embedding_partition_to_used_mem_bytes[partition] = used_mem_bytes
```

Comment:
These 4 lines are repeated 3 times, probably want to factor them out into a function and call that